README.md

```diff
@@ -2,70 +2,187 @@
 
 [![ansible-lint.yml](https://github.com/linux-system-roles/bootloader/actions/workflows/ansible-lint.yml/badge.svg)](https://github.com/linux-system-roles/bootloader/actions/workflows/ansible-lint.yml) [![ansible-test.yml](https://github.com/linux-system-roles/bootloader/actions/workflows/ansible-test.yml/badge.svg)](https://github.com/linux-system-roles/bootloader/actions/workflows/ansible-test.yml) [![markdownlint.yml](https://github.com/linux-system-roles/bootloader/actions/workflows/markdownlint.yml/badge.svg)](https://github.com/linux-system-roles/bootloader/actions/workflows/markdownlint.yml) [![woke.yml](https://github.com/linux-system-roles/bootloader/actions/workflows/woke.yml/badge.svg)](https://github.com/linux-system-roles/bootloader/actions/workflows/woke.yml)
 
-An Ansible role for bootloader and kernel command line management.
+An Ansible role for boot loader and kernel command line management.
 
 ## Requirements
 
-Any pre-requisites that may not be covered by Ansible itself or the role should
-be mentioned here. For instance, if the role uses the EC2 module, it may be a
-good idea to mention in this section that the `boto` package is required.
+None
 
 ## Role Variables
 
-A description of all input variables (i.e. variables that are defined in
-`defaults/main.yml`) for the role should go here as these form an API of the
-role.
+### bootloader_gather_facts
 
-Variables that are not intended as input, like variables defined in
-`vars/main.yml`, variables that are read from other roles and/or the global
-scope (ie. hostvars, group vars, etc.) can be also mentioned here but keep in
-mind that as these are probably not part of the role API they may change during
-the lifetime.
+Whether to gather `bootloader_facts` that contain bot information for all kernels.
 
-Example of setting the variables:
+Default: `false`
+
+Type: `bool`
+
+### bootloader_settings
+
+With this variable, list kernels and their command line parameters that you want to set.
+
+You can specify one or more of the following kernels:
+1. `<kernel-path>` a specific kernel path to create or update with the configuration provided.
+2. `DEFAULT` to update the default entry.
+3. `ALL` to update all of the entries.
+Specifying  or `ALL` updates the default entry and all of the entries, respectively.
+
+Each kernel `dict` might has one or more of the following keys:
+
+* `name` - The name of the setting. `name` is omitted when using `replaced`.
+* `value` - The value for the setting. You must omit `value` if the setting has no value, e.g. `quiet`.
+* `state` - `present` (default) or `absent`. The value `absent` means to remove a setting with `name` name - name must be provided.
+* `previous` - Optional - the only value is `replaced` - this is used to specify that the previous settings should be replaced with the given settings.
+
+Example:
 
 ```yaml
-bootloader_foo: "oof"
-bootloader_bar: "baz"
+bootloader_settings:
+  /boot/vmlinuz-0-rescue-1:
```

Comment:
There are a few ways to specify a kernel.  If I look at the output of grubby on my F38 system:
```
index=0
kernel="/boot/vmlinuz-6.5.8-200.fc38.x86_64"
args="ro rootflags=subvol=root rd.luks.uuid=luks-58c87c49-4d0e-4cd1-a066-69a51238bfac rhgb quiet rd.luks.key=58c87c49-4d0e-4cd1-a066-69a51238bfac=/keyfile:UUID=831a957a-f42d-4ff3-848b-9ac54732816b"
root="UUID=91157ca3-f494-45ed-9f88-d0472e719b17"
initrd="/boot/initramfs-6.5.8-200.fc38.x86_64.img"
title="Fedora Linux (6.5.8-200.fc38.x86_64) 38 (Workstation Edition)"
id="f19756c27f6849d98e70b6cb0c5a0e00-6.5.8-200.fc38.x86_64"
index=1
kernel="/boot/vmlinuz-6.5.7-200.fc38.x86_64"
args="ro rootflags=subvol=root rd.luks.uuid=luks-58c87c49-4d0e-4cd1-a066-69a51238bfac rhgb quiet rd.luks.key=58c87c49-4d0e-4cd1-a066-69a51238bfac=/keyfile:UUID=831a957a-f42d-4ff3-848b-9ac54732816b"
root="UUID=91157ca3-f494-45ed-9f88-d0472e719b17"
initrd="/boot/initramfs-6.5.7-200.fc38.x86_64.img"
title="Fedora Linux (6.5.7-200.fc38.x86_64) 38 (Workstation Edition)"
id="f19756c27f6849d98e70b6cb0c5a0e00-6.5.7-200.fc38.x86_64"
...
```

looks like I can specify a kernel by `index`, `kernel`, `title`, or `id`.  And - it looks like index `0` is the current one, `1` is the previous one, `2` is the one before `1`, etc.  It seems like using `index` is the most simple way.

How about something like this?

```yaml
bootloader_settings:
  - kernel:  # used to specify the kernel by index, path, title, or id criteria - only 1 criteria at a time may be used
      path: /boot/vmlinuz-6.5.7-200.fc38.x86_64
      # or index: 0
      # or title: Fedora Linux (6.5.7-200.fc38.x86_64) 38 (Workstation Edition)
      # or id: f19756c27f6849d98e70b6cb0c5a0e00-6.5.7-200.fc38.x86_64
      # the criteria could also be a list e.g. index: [0, 2, 4] - apply only to kernels at these indexes
  # - kernel: ALL - special value meaning to apply settings to all kernels
    options:
      - name: console
        value: tty0
      - name: print-fatal-signals
        value: 1
      - name: no_timer_check
        state: present
      - name: quiet
      - name: debug
      - previous: replaced
```

wdyt?

Reply:
Yeah ,this is a proper way to get define kernel. One correction is that you cannot use `id`, only path, inde, and title.

I used kernel_index, kernel_path, kernel_title because apparently `bootloader_settings.kernel.index` is used by ansible, so when `bootloader_settings.kernel.path` is set, bootloader_settings.kernel.index is defined and contains `built-in method index of str object`.

I implemented it with Ansible, without an Ansible module, and it is pretty messy, see loop_kernels.yml used to loop inside a loop.